src/java/picard/sam/MergeSamFiles.java

```diff
@@ -83,6 +90,9 @@
     @Option(doc = "Comment(s) to include in the merged output file's header.", optional = true, shortName = "CO")
     public List<String> COMMENT = new ArrayList<String>();
 
+    @Option(shortName = "INTERVALS", doc = "An interval list file that contains the locations of the positions to merge. Assume bam are sorted and indexed.", optional = true)
```

Comment:
My initial idea was to quickly create a merged bam for a given (set of) regions before calling SNPs

> and the resulting mess that would arise from that

Doesn't MergingSamRecordIterator handle this problem ? if not, you're absolutely right, that would be a problem.  :-(

feel free to close that pull request if this is the case.